src/scripts/main.js

```diff
@@ -1,3 +1,30 @@
 'use strict';
 
-// write code here
+const ul = document.querySelector('ul');
+const list = [...ul.querySelectorAll('li')];
+
+function toNumber(str) {
+  return +str.replace(/['$']/g, '').replace(/[',']/g, '.');
+}
+
+function sortList() {
+  list.sort(
+    (li1, li2) => toNumber(li2.dataset.salary) - toNumber(li1.dataset.salary),
```

Comment:
The sorting function assumes that the `data-salary` attribute is always present and correctly formatted. Ensure that this attribute is always available and contains valid data to prevent potential errors.